### PR TITLE
4 Key Macropad + Encoder

### PR DESCRIPTION
**Pull request by mexsistor**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mexsistor/ludmila/config.h [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID    0x2808
+#define PRODUCT_ID   0x0203
```
Comment (drashna) [inline keyboards/mexsistor/ludmila/config.h:24]: This vendor ID is already in use officially.  Please choose a different VID.
https://www.the-sz.com/products/usbid/index.php?v=0x2808&p=&n=

for instance:
```suggestion
#define VENDOR_ID    0x69CC
#define PRODUCT_ID   0x6BF6
```

File: keyboards/mexsistor/ludmila/ludmila.h [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ * The second converts the arguments into a two-dimensional array which
+ * represents the switch matrix.
+ */
+#define LAYOUT( \
+    k00, k01, k20, \
+    k10, k11 \
+) { \
+    { k00, k01, }, \
+    { k10, k11, }, \
+    { k20, ___, }  \
```
Comment (drashna) [inline keyboards/mexsistor/ludmila/ludmila.h:36]:
```suggestion
    { k00, k01 }, \
    { k10, k11 }, \
    { k20, ___ }  \
```

**Review by drashna — CHANGES_REQUESTED** [review]
Is there a specific reason for the custom matrix here?

File: keyboards/mexsistor/ludmila/ludmila.h (hunk 2 above)
Comment (mexsistor) [reply keyboards/mexsistor/ludmila/ludmila.h]: There is no special reason to leave that part there, I use it as a kind of "mapping" when assigning my macros. But you can remove it if you want :) Thank you for your review and your comments.

File: keyboards/mexsistor/ludmila/matrix.c [hunk 3]
```diff
@@ -0,0 +1,258 @@
+/*
```
Comment (fauxpark) [inline keyboards/mexsistor/ludmila/matrix.c:1]: This should be converted to [Custom Matrix Lite](https://docs.qmk.fm/#/custom_matrix?id=lite)

File: keyboards/mexsistor/ludmila/readme.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# ludmila
+
+![ludmila](https://imgur.com/a/QnQTltS)
+
+Ludmila is a 4 Key Macropad with one rotary encoder.
+
+* Keyboard Maintainer: [Kevin M.](https://github.com/mexsistor)
+* Hardware Supported: Ludmila PCB (atmega32u4)
+* Hardware Availability: [Kevin M:](https://www.mexsistor.com/)
+Make example for this keyboard (after setting up your build environment):
```
Comment (fauxpark) [inline keyboards/mexsistor/ludmila/readme.md:10]:
```suggestion

Make example for this keyboard (after setting up your build environment):
```

File: keyboards/mexsistor/ludmila/rules.mk [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
+ENCODER_ENABLE = yes        # Rotary Encoder support
+TAP_DANCE_ENABLE = no      # Support for tap dancing
```
Comment (fauxpark) [inline keyboards/mexsistor/ludmila/rules.mk:29]:
```suggestion
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
ENCODER_ENABLE = yes        # Rotary Encoder support
CUSTOM_MATRIX = yes
```

File: keyboards/mexsistor/ludmila/rules.mk [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+SRC = matrix.c
```
Comment (fauxpark) [inline keyboards/mexsistor/ludmila/rules.mk:31]:
```suggestion
SRC += matrix.c
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/mexsistor/ludmila/rules.mk [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = caterina
+
+# Build Options
+#   change yes to no to disable
+#
+
+CUSTOM_MATRIX = lite			# Custom scanning of matrix
```
Comment (fauxpark) [inline keyboards/mexsistor/ludmila/rules.mk:11]:
```suggestion
CUSTOM_MATRIX = yes			# Custom scanning of matrix
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/mexsistor/ludmila/rules.mk [hunk 8]
```diff
@@ -0,0 +1,27 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = caterina
+
+# Build Options
+#   change yes to no to disable
+#
+
```
Comment (fauxpark) [inline keyboards/mexsistor/ludmila/rules.mk:10]:
```suggestion
```

File: keyboards/mexsistor/ludmila/readme.md [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+
+![ludmila](https://imgur.com/a/QnQTltS)
+
+Ludmila is a 4 Key Macropad with one rotary encoder.
+
+* Keyboard Maintainer: [Kevin M.](https://github.com/mexsistor)
+* Hardware Supported: Ludmila PCB (atmega32u4)
+* Hardware Availability: [Kevin M:](https://www.mexsistor.com/)
+
+
```
Comment (fauxpark) [inline keyboards/mexsistor/ludmila/readme.md:11]:
```suggestion
```

File: keyboards/mexsistor/ludmila/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
```
Comment (fauxpark) [inline keyboards/mexsistor/ludmila/keymaps/default/keymap.c:16]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by fauxpark** [comment]
@mexsistor please submit the other keyboard in a different PR.

**Pull request closed without merging**